liam2/alignment.py

```diff
@@ -340,6 +340,7 @@ def compute(self, context, score, need=None,
             # calculate without leave_filter and without take_filter
             if need is None:
                 need = sum(score)
+            need = np.floor(need).astype(int)

```

Comment:
Wouldn't it be better to handle this as another option for frac_need? (e.g. "ignore" or "floor") and possibly use that option by default for sidewalk, if it makes more sense (which I am not convinced of).